disperser/batcher/grpc/dispatcher.go

```diff
@@ -34,23 +34,27 @@ func NewDispatcher(cfg *Config, logger common.Logger) *dispatcher {
 
 var _ disperser.Dispatcher = (*dispatcher)(nil)
 
-func (c *dispatcher) DisperseBatch(ctx context.Context, state *core.IndexedOperatorState, blobs []core.EncodedBlob, header *core.BatchHeader) chan core.SignerMessage {
+func (c *dispatcher) DisperseBatch(ctx context.Context, state *core.IndexedOperatorState, blobs []core.EncodedBlob, batchHeader *core.BatchHeader) chan core.SignerMessage {
 	update := make(chan core.SignerMessage, len(state.IndexedOperators))
 
 	// Disperse
-	c.sendAllChunks(ctx, state, blobs, header, update)
+	c.sendAllChunks(ctx, state, blobs, batchHeader, update)
 
 	return update
 }
 
-func (c *dispatcher) sendAllChunks(ctx context.Context, state *core.IndexedOperatorState, blobs []core.EncodedBlob, header *core.BatchHeader, update chan core.SignerMessage) {
+func (c *dispatcher) sendAllChunks(ctx context.Context, state *core.IndexedOperatorState, blobs []core.EncodedBlob, batchHeader *core.BatchHeader, update chan core.SignerMessage) {
 	for id, op := range state.IndexedOperators {
 		go func(op core.IndexedOperatorInfo, id core.OperatorID) {
-			blobMessages := make([]*core.BlobMessage, len(blobs))
-			for i, blob := range blobs {
-				blobMessages[i] = blob[id]
+			blobMessages := make([]*core.BlobMessage, 0)
+			for _, blob := range blobs {
+				blobMessages = append(blobMessages, &core.BlobMessage{
+					BlobHeader: blob.BlobHeader,
+					// Bundles will be empty if the operator is not in the quorums blob is dispersed on
+					Bundles: blob.BundlesByOperator[id],
+				})
 			}
-			sig, err := c.sendChunks(ctx, blobMessages, header, &op)
+			sig, err := c.sendChunks(ctx, blobMessages, batchHeader, &op)
```

Comment:
Why disperse to an operator if it's not in any quorum of a batch?

Reply:
It should only disperse the blob headers to the operators not in the quorums. An operator needs all the blob headers (even the ones they don't participate on) to verify the batch root

Reply:
Why do they need to verify the batch root? If the batch has quorum 1,2 (aggregating from blobs inside), and an operator is only in quorum 3, why it has to verify/sign a batch that it has no responsibility?